tests/Service/ConfigurationFactoryTest.php

```diff
@@ -47,7 +48,7 @@ public function testWillInstantiateConfigWithoutNamingStrategySetting(): void
             ],
         ];
         $this->serviceManager->setService('config', $config);
-        $ormConfig = $this->factory->createService($this->serviceManager);
+        $ormConfig = $this->factory->__invoke($this->serviceManager, Configuration::class);
```

Comment:
Calling __invoke directly is unusual.  I recommend 
```php
$ormConfig = $this->factory($this->serviceManager, Configuration::class);
```

Reply:
You cannot use `$this->factory()` here, as then PHP tries to call `ConfigurationFactoryTest::factory()` instead of `ConfigurationFactoryTest::$factory()`. I could use `($this->factory)()` though, which still looks a bit akward to me.

Reply:
I think `($this->factory)()` is perfectly acceptable.  I resolve objects this way all the time and that's especially true in Doctrine.

```php
$entity = (new Entity()) 
    ->setName('name')
    ;
```

is a common pattern in my work.  I certainly think `($this->factory)()` is much more readable than directly calling `__invoke`.

Reply:
Added with 46b6dc2, requesting approval 😉